plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -256,6 +266,24 @@ else if (upperBound != null) {
         return Domain.create(ValueSet.ofRanges(statisticsRange), containsNulls);
     }
 
+    @VisibleForTesting
+    static boolean partitionMatchesConstraint(
+            Set<IcebergColumnHandle> identityPartitionColumns,
+            Map<Integer, Optional<String>> partitionKeys,
+            ConstraintEvaluator constraintEvaluator)
+    {
```

Comment:
if `constraintEvaluator.getArguments() intersect identityPartitionColumns is empty`, we should exit early here.

Reply:
```
if (intersection(constraintEvaluator.getArguments(), identityPartitionColumns).isEmpty()) {
    return true;
}
```
and do not use `else`